File: src/main/java/arcana/common/items/ModItems.java
package arcana.common.items;

import arcana.common.blocks.ModBlocks;
import arcana.common.items.wand.*;
import net.minecraft.item.BlockItem;
import net.minecraft.item.Item;

import static arcana.utils.Util.arcLoc;

public class ModItems {
    public static Item Arcanum = new ArcanumItem();
    public static Item FIREWAND = new BasicWand();
    public static Item RESEARCH_TABLE = new BlockItem(ModBlocks.RESEARCH_TABLE_RIGHT, ArcanaGroup.itemProps()).setRegistryName("research_table");
    public static Item SCRIBING_TOOLS = new Item(ArcanaGroup.itemProps().durability(100).setNoRepair()).setRegistryName("scribing_tools");
    public static Item RESEARCH_NOTE = new ResearchNote(ArcanaGroup.itemProps(), false).setRegistryName("research_note");
    public static Item RESEARCH_NOTE_COMPLETE = new ResearchNote(ArcanaGroup.itemProps(), true).setRegistryName("research_note_complete");

    public static WandItem WAND = new WandItem();
    //public static Item STAFF = new StaffItem();
    public static Item GAUNTLET = new Gauntlet();
    public static FocusItem FOCUS = new FocusItem(arcLoc("focus_1"));

    public static CapItem AMBER_CAP = new CapItem(arcLoc("amber_cap"), 50, 30, 2);
    public static CapItem BAMBOO_CAP = new CapItem(arcLoc("bamboo_cap"), 50, 30, 2);
    public static CapItem CLAY_CAP = new CapItem(arcLoc("clay_cap"), 32767, 32767, -1);
    public static CapItem ELDRITCH_CAP = new CapItem(arcLoc("eldritch_cap"), 70, 50, 2);
    public static CapItem HONEY_CAP = new CapItem(arcLoc("honey_cap"), 50, 30, 2);
    public static CapItem COPPER_CAP = new CapItem(arcLoc("copper_cap"), 15, 5, 2);
    public static CapItem ELEMENTIUM_CAP = new CapItem(arcLoc("elementium_cap"), 0, 0, 0);
    public static CapItem GOLD_CAP = new CapItem(arcLoc("gold_cap"), 25, 15, 2);
    public static CapItem IRON_CAP = new CapItem(arcLoc("iron_cap"), 10, 3, 1);
    public static CapItem MANASTEEL_CAP = new CapItem(arcLoc("manasteel_cap"), 0, 0, 0);
    public static CapItem SILVER_CAP = new CapItem(arcLoc("silver_cap"), 0, 75, 2);
    public static CapItem TERRASTEEL_CAP = new CapItem(arcLoc("terrasteel_cap"), 0, 0, 0);
    public static CapItem THAUMIUM_CAP = new CapItem(arcLoc("thaumium_cap"), 50, 25, 3);
    public static CapItem VOID_CAP = new CapItem(arcLoc("void_cap"), 75, 45, 3);
    public static CapItem LEATHER_CAP = new CapItem(arcLoc("leather_cap"), 50, 30, 2);
    public static CapItem MECHANICAL_CAP = new CapItem(arcLoc("mechanical_cap"), 60, 40, 2);
    public static CapItem PRISMARINE_CAP = new CapItem(arcLoc("prismarine_cap"), 50, 30, 2);
    public static CapItem QUARTZ_CAP = new CapItem(arcLoc("quartz_cap"), 50, 30, 2);
    public static CapItem SHULKER_CAP = new CapItem(arcLoc("shulker_cap"), 0, 35, 2);
    public static CapItem NETHERITE_CAP = new CapItem(arcLoc("netherite_cap"), 65, 40, 3);

    public static CoreItem WOOD_WAND_CORE = new CoreItem(arcLoc("wood_wand_core"), 5, 1, 1);
    public static CoreItem GREATWOOD_WAND_CORE = new CoreItem(arcLoc("greatwood_wand_core"), 25, 3, 2);
    public static CoreItem TAINTED_WAND_CORE = new CoreItem(arcLoc("tainted_wand_core"), 200, 30, 2);
    public static CoreItem DAIR_WAND_CORE = new CoreItem(arcLoc("dair_wand_core"), 10, 1, 3);
    public static CoreItem HAWTHORN_WAND_CORE = new CoreItem(arcLoc("hawthorn_wand_core"), 15, 1, 3);
    public static CoreItem SILVERWOOD_WAND_CORE = new CoreItem(arcLoc("silverwood_wand_core"), 100, 5, 4);
    public static CoreItem ARCANIUM_WAND_CORE = new CoreItem(arcLoc("arcanium_wand_core"), 150, 10, 4);
    public static CoreItem BLAZE_WAND_CORE = new CoreItem(arcLoc("blaze_wand_core"), 199, 10, 2);
    public static CoreItem ENDROD_WAND_CORE = new CoreItem(arcLoc("endrod_wand_core"), 225, 15, 4);
    public static CoreItem BONE_WAND_CORE = new CoreItem(arcLoc("bone_wand_core"), 100, 5, 2);
    public static CoreItem ICE_WAND_CORE = new CoreItem(arcLoc("ice_wand_core"), 100, 10, 2);
    public static CoreItem ARCANE_STONE_WAND_CORE = new CoreItem(arcLoc("arcane_stone_wand_core"), 100, 10, 2);
    public static CoreItem OBSIDIAN_WAND_CORE = new CoreItem(arcLoc("obsidian_wand_core"), 100, 5, 2);
    public static CoreItem SUGAR_CANE_WAND_CORE = new CoreItem(arcLoc("sugar_cane_wand_core"), 100, 10, 2);
    public static CoreItem MECHANICAL_WAND_CORE = new CoreItem(arcLoc("mechanical_wand_core"), 125, 15, 2);
    public static CoreItem ELDRITCH_WAND_CORE = new CoreItem(arcLoc("eldritch_wand_core"), 250, 30, 2);
    public static CoreItem CLAY_WAND_CORE = new CoreItem(arcLoc("clay_wand_core"), Short.MAX_VALUE, Short.MAX_VALUE, -1);
}

File: src/main/kotlin/arcana/common/items/ModItems.kt
package arcana.common.items

import arcana.common.blocks.ModBlocks
import arcana.common.items.wand.*
import arcana.utils.Util.arcLoc
import net.minecraft.item.BlockItem
import net.minecraft.item.Item


object ModItems {
    private fun <T : Item> item(name: String, item: T) = item.setRegistryName(arcLoc(name)) as T

    @JvmField val ARCANUM = item("arcanum", ArcanumItem())
    @JvmField val FIREWAND = item("basicwand", BasicWand())
    @JvmField val RESEARCH_TABLE = item("research_table", BlockItem(ModBlocks.RESEARCH_TABLE_RIGHT, ArcanaGroup.props))
    @JvmField val SCRIBING_TOOLS = item("scribing_tools", Item(ArcanaGroup.props.durability(100).setNoRepair()))
    @JvmField val RESEARCH_NOTE = item("research_note", ResearchNote(ArcanaGroup.props, false))
    @JvmField val RESEARCH_NOTE_COMPLETE = item("research_note_complete",
                                          ResearchNote(ArcanaGroup.props, true))

    @JvmField val WAND = item("wand", WandItem())
    @JvmField val STAFF = item("staff", StaffItem())
    @JvmField val GAUNTLET = item("gauntlet", Gauntlet())

    @JvmField val FOCUS = item("focus", FocusItem())
    //@JvmField val AMBER_CAP = CapItem(arcLoc("amber"), 50, 30, 2)
    //@JvmField val BAMBOO_CAP = CapItem(arcLoc("bamboo"), 50, 30, 2)
    //@JvmField val CLAY_CAP = CapItem(arcLoc("clay"), 32767, 32767, -1)
    // @JvmField val ELDRITCH_CAP = CapItem(arcLoc("eldritch"), 70, 50, 2)
    //@JvmField val HONEY_CAP = CapItem(arcLoc("honey"), 50, 30, 2)
    //@JvmField val COPPER_CAP = CapItem(arcLoc("copper"), 15, 5, 2)
    //@JvmField val ELEMENTIUM_CAP = CapItem(arcLoc("elementium"), 0, 0, 0)
    @JvmField val GOLD_CAP = CapItem(arcLoc("gold"), 25, 15, 2)
    @JvmField val IRON_CAP = CapItem(arcLoc("iron"), 10, 3, 1)
    //@JvmField val MANASTEEL_CAP = CapItem(arcLoc("manasteel"), 0, 0, 0)
    @JvmField val SILVER_CAP = CapItem(arcLoc("silver"), 0, 75, 2)
    //@JvmField val TERRASTEEL_CAP = CapItem(arcLoc("terrasteel"), 0, 0, 0)
    @JvmField val THAUMIUM_CAP = CapItem(arcLoc("thaumium"), 50, 25, 3)
    @JvmField val VOID_CAP = CapItem(arcLoc("void"), 75, 45, 3)
    ///@JvmField val LEATHER_CAP = CapItem(arcLoc("leather"), 50, 30, 2)
    //@JvmField val MECHANICAL_CAP = CapItem(arcLoc("mechanical"), 60, 40, 2)
    //@JvmField val PRISMARINE_CAP = CapItem(arcLoc("prismarine"), 50, 30, 2)
    //@JvmField val QUARTZ_CAP = CapItem(arcLoc("quartz"), 50, 30, 2)
    //@JvmField val SHULKER_CAP = CapItem(arcLoc("shulker"), 0, 35, 2)
    //@JvmField val NETHERITE_CAP = CapItem(arcLoc("netherite"), 65, 40, 3)

    @JvmField val WOOD_CORE = CoreItem(arcLoc("wood"), 5, 1, 1)
    @JvmField val GREATWOOD_CORE = CoreItem(arcLoc("greatwood"), 25, 3, 2)
    //@JvmField val TAINTED_CORE = CoreItem(arcLoc("tainted"), 200, 30, 2)
    @JvmField val DAIR_CORE = CoreItem(arcLoc("dair"), 10, 1, 3)
    //@JvmField val HAWTHORN_CORE = CoreItem(arcLoc("hawthorn"), 15, 1, 3)
    @JvmField val SILVERWOOD_CORE = CoreItem(arcLoc("silverwood"), 100, 5, 4)
    @JvmField val ARCANIUM_CORE = CoreItem(arcLoc("arcanium"), 150, 10, 4)
    @JvmField val BLAZE_CORE = CoreItem(arcLoc("blaze"), 199, 10, 2)
    //@JvmField val ENDROD_CORE = CoreItem(arcLoc("endrod"), 225, 15, 4)
    //@JvmField val BONE_CORE = CoreItem(arcLoc("bone"), 100, 5, 2)
    @JvmField val ICE_CORE = CoreItem(arcLoc("ice"), 100, 10, 2)
    //@JvmField val ARCANE_STONE_CORE = CoreItem(arcLoc("arcane_stone"), 100, 10, 2)
    //@JvmField val OBSIDIAN_CORE = CoreItem(arcLoc("obsidian"), 100, 5, 2)
    //@JvmField val SUGAR_CANE_CORE = CoreItem(arcLoc("sugar_cane"), 100, 10, 2)
    //@JvmField val MECHANICAL_CORE = CoreItem(arcLoc("mechanical"), 125, 15, 2)
    //@JvmField val ELDRITCH_CORE = CoreItem(arcLoc("eldritch"), 250, 30, 2)
    //@JvmField val CLAY_CORE = CoreItem(arcLoc("clay"), Short.MAX_VALUE.toInt(), Short.MAX_VALUE.toInt(), -1)
}